test/e2e/storage/vsphere_stress.go

```diff
@@ -0,0 +1,175 @@
+package storage
+
+import (
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/api/core/v1"
+	storageV1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	k8stype "k8s.io/apimachinery/pkg/types"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/pkg/cloudprovider/providers/vsphere"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"os"
+	"strconv"
+	"sync"
+)
+
+/*
+	Induce stress to create volumes in parallel with multiple threads based on user configurable values for number of threads and iterations per thread.
+	The following actions will be performed as part of this test.
+
+	1. Create Storage Classes of 4 Categories (Default, SC with Non Default Datastore, SC with SPBM Policy, SC with VSAN Storage Capalibilies.)
+	2. READ VCP_STRESS_INSTANCES, VCP_STRESS_ITERATIONS, VSPHERE_SPBM_POLICY_NAME and VSPHERE_DATASTORE from System Environment.
+	3. Launch goroutine for volume lifecycle operations.
+	4. Each instance of routine iterates for n times, where n is read from system env - VCP_STRESS_ITERATIONS
+	5. Each iteration creates 1 PVC, 1 POD using the provisioned PV, Verify disk is attached to the node, Verify pod can access the volume, delete the pod and finally delete the PVC.
+*/
+var _ = SIGDescribe("vsphere cloud provider stress [Feature:vsphere]", func() {
+	f := framework.NewDefaultFramework("vcp-stress")
+	const (
+		volumesPerNode = 55
```

Comment:
Where is this actually used?

Reply:
This is used at L55
```
Expect(instances <= volumesPerNode*len(nodeList.Items)).To(BeTrue(), fmt.Sprintf("Number of Instances should be less or equal: %v", volumesPerNode*len(nodeList.Items)))

Reply:
But where are you actually creating 55 volumes per node?

Reply:
This is just to control the number of instances test should allow to start. If let’s say we have 1 ready and schedulable node, then test should not allow to launch more than 55 instances.

Test relies on the kubernetes scheduler and not deciding the scheduling based on number volumes attached to the node.